app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt

```diff
@@ -41,58 +46,84 @@ import com.google.samples.apps.nowinandroid.feature.foryou.navigation.ForYouDest
 import com.google.samples.apps.nowinandroid.feature.interests.R as interestsR
 import com.google.samples.apps.nowinandroid.feature.interests.navigation.InterestsDestination
 import com.google.samples.apps.nowinandroid.navigation.TopLevelDestination
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 
 @Composable
 fun rememberNiaAppState(
     windowSizeClass: WindowSizeClass,
+    networkMonitor: NetworkMonitor,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
     navController: NavHostController = rememberNavController()
 ): NiaAppState {
     NavigationTrackingSideEffect(navController)
-    return remember(navController, windowSizeClass) {
-        NiaAppState(navController, windowSizeClass)
+    return remember(navController, coroutineScope, windowSizeClass, networkMonitor) {
+        NiaAppState(navController, coroutineScope, windowSizeClass, networkMonitor)
     }
 }
 
 @Stable
 class NiaAppState(
     val navController: NavHostController,
-    val windowSizeClass: WindowSizeClass
+    val coroutineScope: CoroutineScope,
+    val windowSizeClass: WindowSizeClass,
+    networkMonitor: NetworkMonitor,
 ) {
     val currentDestination: NavDestination?
         @Composable get() = navController
             .currentBackStackEntryAsState().value?.destination
 
+    private var _shouldShowSettingsDialog by mutableStateOf(false)
+    val shouldShowSettingsDialog
+        get() = _shouldShowSettingsDialog
+
     val shouldShowBottomBar: Boolean
         get() = windowSizeClass.widthSizeClass == WindowWidthSizeClass.Compact ||
             windowSizeClass.heightSizeClass == WindowHeightSizeClass.Compact
 
     val shouldShowNavRail: Boolean
         get() = !shouldShowBottomBar
 
+    val isOffline = networkMonitor.isOnline
+        .map(Boolean::not)
+        .stateIn(
+            scope = coroutineScope,
+            started = SharingStarted.WhileSubscribed(5_000),
+            initialValue = false
+        )
+
     /**
-     * Top level destinations to be used in the BottomBar and NavRail
+     * Map of top level destinations to be used in the TopBar, BottomBar and NavRail. The key is the
+     * route.
      */
-    val topLevelDestinations: List<TopLevelDestination> = listOf(
-        TopLevelDestination(
+    val topLevelDestinations: Map<String, TopLevelDestination> = mapOf(
+        ForYouDestination.route to TopLevelDestination(
```

Comment:
Why do you need the route? That'd force all routes to be public and exposed to the `app` module

Reply:
All routes are already public, they were just exposed as a `List` rather than `Map`. This is necessary so that the bottom bar can render all the destinations, and the top bar can obtain the screen title. 

The reason for changing to a `Map` is that it's O(1) to obtain the current top level destination, rather than O(N) to iterate through the list. 

Happy to discuss better ideas, it does feel a bit messy.

Reply:
In the interest of getting this merged I'm going to leave as-is. Please lmk if you feel strongly about changing this (it might change with your nav change anyway).